frontend/src/metabase/nav/containers/MainNavbar/tests/setup.tsx

```diff
@@ -0,0 +1,223 @@
+    id: 1,
+    name: "Sample Database",
+    is_sample: true,
+    can_upload: user?.is_superuser || (isUploadEnabled && hasDataAccess),
+  });
+
+  const USER_DATABASE = createMockDatabase({
+    id: 2,
+    name: "User Database",
+    is_sample: false,
+  });
+
+  const databases = [];
+
+  if (hasDataAccess) {
+    databases.push(SAMPLE_DATABASE);
+
+    if (withAdditionalDatabase) {
+      databases.push(USER_DATABASE);
+    }
+  }
+  const OUR_ANALYTICS = createMockCollection({
+    ...ROOT_COLLECTION,
+    can_write: user?.is_superuser || canCurateRootCollection,
+  });
+
+  const collections = [TEST_COLLECTION];
+
+  const personalCollection = user
+    ? createMockCollection({
+        ...PERSONAL_COLLECTION_BASE,
+        personal_owner_id: user.id,
+      })
+    : null;
+
+  if (personalCollection && user) {
+    user.personal_collection_id = 1;
+    collections.push(personalCollection);
+  }
+
+  setupCollectionsEndpoints({
+    collections,
+    rootCollection: OUR_ANALYTICS,
+  });
+  setupCollectionByIdEndpoint({
+    collections: [PERSONAL_COLLECTION_BASE, TEST_COLLECTION],
+  });
+  setupDatabasesEndpoints(databases);
+  setupSearchEndpoints(models);
+  setupCollectionItemsEndpoint({
+    collection: createMockCollection(OUR_ANALYTICS),
+    collectionItems: [],
+  });
+  fetchMock.get("path:/api/bookmark", []);
+
+  if (openQuestionCard) {
+    setupCardsEndpoints([openQuestionCard]);
+  }
+
+  let dashboardId: DashboardId | null = null;
+  const dashboardsForState: DashboardState["dashboards"] = {};
+  const dashboardsForEntities: Dashboard[] = [];
+  if (openDashboard) {
+    dashboardId = openDashboard.id;
+    dashboardsForState[openDashboard.id] = {
+      ...openDashboard,
+      dashcards: openDashboard.dashcards.map(c => c.id),
+    };
+    dashboardsForEntities.push(openDashboard);
+  }
+
+  const storeInitialState = createMockState({
+    currentUser: user,
+    dashboard: createMockDashboardState({
+      dashboardId,
+      dashboards: dashboardsForState,
+    }),
+    qb: createMockQueryBuilderState({ card: openQuestionCard }),
+    entities: createMockEntitiesState({ dashboards: dashboardsForEntities }),
+    settings: createMockSettingsState({
+      "uploads-settings": {
+        db_id: hasDWHAttached || isUploadEnabled ? SAMPLE_DATABASE.id : null,
+        schema_name: null,
+        table_prefix: null,
+      },
+      "instance-creation": instanceCreationDate,
+      "token-features": createMockTokenFeatures({
+        attached_dwh: hasDWHAttached,
+      }),
+    }),
+  });
+
+  if (hasEnterprisePlugins) {
+    setupEnterprisePlugins();
+  }
+
+  renderWithProviders(
+    <Route
+      path={route}
+      component={props => <MainNavbar {...props} isOpen />}
+    />,
+    {
+      storeInitialState,
+      initialRoute: pathname,
+      withRouter: true,
+      withDND: true,
+    },
+  );
+
+  await waitForLoaderToBeRemoved();
+  await waitForLoaderToBeRemoved(); // tests will fail without the 2nd call
```

Comment:
I think this might be due to the second loading indicator, specific to Browse models